cinema/models.py

```diff
@@ -150,7 +263,122 @@ class Meta:
         ordering = ["user__first_name", "user__last_name"]
 
 
+class FilmManager(models.Manager):
+    """
+    Custom Manager, adding extra manager methods.
+
+    Custom Manager used in 'Film' model by extending base Manager class
+    and instantiating custom Manager in 'Film' model.
+    """
+    def get_queryset(self):
+        """
+        Return QuerySet object with cached data of some related models.
+        """
+        return super().get_queryset().select_related(
+            "imdb_rating", "mpaa_rating", "product"
+        )
+
+    def get_brief_films_data(self):
+        """
+        Return QuerySet object with data of specified fields of 'Film' model.
+        """
+        return super().get_queryset().values_list("pk", "title", named=True)
+
+    def filter_by_film_selected_year(self, selected_year):
+        """
+        Return QuerySet object filtered by selected year and order by
+        imdb rating of movies.
+        """
+        return self.get_queryset().filter(
+            release_data__year=selected_year
+        ).order_by("-imdb_rating__value")
+
+    def filter_by_film_selected_genre(self, selected_genre):
+        """
+        Return QuerySet object filtered by selected genre and order by
+        imdb rating of movies.
+        """
+        return self.get_queryset().filter(
+            genre__name=selected_genre
+        ).order_by("-imdb_rating__value")
+
+    def filter_by_film_selected_country(self, selected_country):
```

Comment:
check this name construction everywhere.

Reply:
Changed method name